packages/webapp/src/components/collection/deckPanel.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { useState } from 'react'
+import { Deck, Card } from 'src/store/types' 
+import Image from 'next/image' 
+
+interface DeckConstructionPanelProps {
+    deck: Deck
+    selectedCards: Card[]
+    onCardSelect: (card: Card) => void
+    onSave: (deck: Deck) => void
+    onCancel: () => void
+  }
+  
+  
+  const DeckConstructionPanel : React.FC<DeckConstructionPanelProps> = ({ deck, selectedCards = [], onCardSelect, onSave, onCancel }) => {
+    const [ deckName, setDeckName ] = useState(deck.name)
+    const [ isDeckNameValid, setIsDeckNameValid ] = useState(true)
+
+    const handleDeckNameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+      const newName = event.target.value
+      setDeckName(event.target.value)
+      setIsDeckNameValid(nameValid(newName))
+    }
+  
+    const nameValid = (name: string) => name.trim().length > 0
+
+    const handleSave = () => {
+      if (!nameValid(deckName)) {
+        setIsDeckNameValid(false)
+        return
+      }
+
+      const newDeck = {
+        name: deckName.trim(),
+        cards: selectedCards
+      }
+
+      onSave(newDeck)
+    }
+
+  return (
+    <div className="w-full flex flex-col items-center p-3">
+      <div className="flex justify-between items-center">
+        {/* Deck Name Input */}
+        <input
+          type="text"
+          value={deckName}
+          onChange={handleDeckNameChange}
+          style={{ outline: isDeckNameValid ? "none" : "2px solid red" }}
+          className="flex-grow px-2 py-2 border rounded-md text-gray-100 focus:outline-none focus:ring-2 focus:ring-blue-500 focus:border-transparent m-1.5"
+          placeholder=" Deck name"
+        />

```

Comment:
The input font color here is a very light gray, it's impossible to read. Just a black or very dark gray will do. The placeholder font is a darker grey, that's fine (but if you use grey for the text, the difference has to be noticeable ... honestly just do black, black is fine here).